### PR TITLE
 Move Dataset API from telemetry-batch-view to its own package on maven

### PR DESCRIPTION
**Pull request by vitillo**
See [Bug 1283446](https://bugzilla.mozilla.org/show_bug.cgi?id=1283446). Since I was at it I completely rewrote the test suite using [fakes3](https://github.com/jubos/fake-s3). I am planning to add CI integration before this gets merged.

### CODE REVIEW

**Comment by mreid-moz** [comment]
Does the Heka-reading code work with the gzipped format a-la [this pr](https://github.com/mozilla/python_moztelemetry/pull/88) and [bug 1302264](https://bugzilla.mozilla.org/show_bug.cgi?id=1302264)?

File: src/test/scala/com/mozilla/telemetry/utils/S3StoreTest.scala [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    S3Store.isPrefixEmpty(bucketName, "a/e/") should be (false)
+    S3Store.deleteKey(bucketName, "a/e/f/")
+    a[Exception] should be thrownBy {
+      S3Store.getKey(bucketName, "a/e/f")
+    }
+    S3Store.isPrefixEmpty(bucketName, "a/e/") should be (true)
+  }
+
+  it can "read gzipped files" in {
+    /* Not supported yet https://github.com/jubos/fake-s3/pull/52
```
Comment (whd) [inline src/test/scala/com/mozilla/telemetry/utils/S3StoreTest.scala]: The referenced PR was closed recently, so perhaps gzip is supported? I am guessing fake-s3 is your answer to the more general testing issues discussed in https://github.com/mozilla/telemetry-batch-view/pull/126.

Comment (vitillo) [inline src/test/scala/com/mozilla/telemetry/utils/S3StoreTest.scala]: I rewrote the testing infrastructure to address the more general testing issues with telemetry-batch-view.

While the Dataset API [supports gzipped files](https://github.com/vitillo/moztelemetry/blob/second/src/main/scala/com/mozilla/telemetry/utils/S3.scala#L24) (it's the same code we are using in telemetry-batch-view) fake-s3 though doesn't support it just yet. In other words we can't write the test for it but we will be able to do so very soon.

**Comment by mreid-moz** [comment]
Can we add a link somewhere - either in the Heka-related code or in the README - to where the ` src/main/protobuf/heka.proto` comes from?

File: src/test/scala/com/mozilla/telemetry/heka/DatasetTest.scala [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+
+import java.io.{ByteArrayInputStream, InputStream}
+
+import com.amazonaws.services.s3.S3ClientOptions
+import com.amazonaws.services.s3.model.ObjectMetadata
+import com.mozilla.telemetry.utils.{ObjectSummary, S3Store}
+import org.apache.spark.{SparkConf, SparkContext}
+import org.scalatest.{BeforeAndAfterAll, FlatSpec, Matchers}
+
+class DatasetTest extends FlatSpec with Matchers with BeforeAndAfterAll{
```
Comment (mreid-moz) [inline src/test/scala/com/mozilla/telemetry/heka/DatasetTest.scala]: nit: please add a space before `{`

File: scalastyle-config.xml [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+ */]]></parameter>
+  </parameters>
+ </check>
+ <check level="warning" class="org.scalastyle.scalariform.SpacesAfterPlusChecker" enabled="true"></check>
+ <check level="warning" class="org.scalastyle.file.WhitespaceEndOfLineChecker" enabled="true"></check>
+ <check level="warning" class="org.scalastyle.scalariform.SpacesBeforePlusChecker" enabled="true"></check>
+ <check level="warning" class="org.scalastyle.file.FileLineLengthChecker" enabled="true">
+  <parameters>
+   <parameter name="maxLineLength"><![CDATA[160]]></parameter>
+   <parameter name="tabSize"><![CDATA[4]]></parameter>
```
Comment (mreid-moz) [inline scalastyle-config.xml]: Indentation in the scala code is all 2 spaces - should we set the tab size to 2 as well?

**Comment by vitillo** [comment]
All done.

**Pull request merged**